src/D_functions_and_procedures.sql

```diff
@@ -146,3 +146,40 @@ BEGIN
     PERFORM rate(john_id, 'tt1375666', 9.0); 
     PERFORM rate(jane_id, 'tt1375666', 7.0); 
 END $$;
+
+-- D7
+CREATE OR REPLACE FUNCTION calculate_name_rating() RETURNS VOID AS $$
+DECLARE
+    name_id INT;
+    weighted_rating DECIMAL(5, 2);
+BEGIN
+    -- Iterate through all persons (actors and crew members)
+    FOR name_id IN (SELECT person_id FROM person) LOOP
```

Comment:
Is it possible to do this without the `FOR LOOP`, because it performs quite bad on larger datasets. Maybe use a with statement?

If its not possible lets use the loop